File: examples/demo-csharp/Calc/Calculator.cs
namespace Calc;

public static class Calculator
{
    public static int Clamp(int value, int lo, int hi)
    {
        if (value < lo) return lo;
        if (value > hi) return hi;
        return value;
    }

    public static double Average(List<double> values)
    {
        if (values.Count == 0) return 0;
        double sum = 0;
        foreach (double v in values)
        {
            sum += v;
        }
        return sum / values.Count;
    }

    public static bool IsEven(int n)
    {
        return n % 2 == 0;
    }

    public static int CountAbove(List<double> values, double threshold)
    {
        int count = 0;
        foreach (double v in values)
        {
            if (v > threshold)
            {
                count += 1;
            }
        }
        return count;
    }

    public static string Describe(int n)
    {
        if (n == 0) return "zero";
        if (n < 0) return "negative";
        return "positive";
    }
}

File: examples/demo-java/src/main/java/calc/Calculator.java
package calc;

import java.util.List;

/** Small library with real behavior for mutation testing. */
public final class Calculator {

    private Calculator() {}

    public static int clamp(int value, int lo, int hi) {
        if (value < lo) return lo;
        if (value > hi) return hi;
        return value;
    }

    public static double average(List<Double> values) {
        if (values.isEmpty()) return 0.0;
        double sum = 0.0;
        for (double v : values) {
            sum += v;
        }
        return sum / values.size();
    }

    public static boolean isEven(int n) {
        return n % 2 == 0;
    }

    public static int countAbove(List<Double> values, double threshold) {
        int count = 0;
        for (double v : values) {
            if (v > threshold) {
                count += 1;
            }
        }
        return count;
    }

    public static String describe(int n) {
        if (n == 0) return "zero";
        if (n < 0) return "negative";
        return "positive";
    }
}
